### DISCUSSION TITLE
TelePeriod funktioniert nicht bei IR-Leseköpfen

### DISCUSSION THREAD

**Discussion by Steve2901** [arendst/Tasmota#21780] [Q&A]
Bei meinem ISKRA 175 Stromzähler werden die Stromdaten mehrfach pro Sekunde aktualisiert (Display).... also so 1-3x.... kann ich nicht genau sagen wie oft....

Problem was ich habe ist, dass mein MQTT Server im Sekundentakt (oder noch öfters) mit den Daten zugebombt wird.
Mit TelePeriod 10....30 etc habe ich das schon versucht, es hat nichts gebracht. Die Daten werden im Sekundentakt an meinen MQTT Server übertragen und da ich im ioBroker das Topic aboniert habe werden die Datenpunkte auch ständig im Sekundentakt aktualisiert. ISt das so gewollt oder ein Bug? Kann man nicht einstellen in welchem Bereich er die Daten Senden soll?

Wäre super wenn es da ne Lösung gäbe. Verwendet wird Tasmota 14.1.0 auf einem Wemos D1 Mini.

Ich danke schonmal herzlichst für die Bearbeitung des Problems.

VG Steve

![grafik](https://github.com/user-attachments/assets/4d019e65-cdec-4f0b-bbe6-aedb398d3dce)
![grafik](https://github.com/user-attachments/assets/552ccff2-5a13-4ffe-bba1-8764f5736ef3)

**Comment by sfromis** [comment]
TelePeriod is for telemetry messages, which are automatically published by Tasmota irrespective of what happens. This does **not** apply to any message having the `tele/` prefix, and IR messages are generally based on something being received.

I don't know if this applies to what you'e seeing, but the meter metrics does have an indicator to force immediate publish (or not) of data, instead of at TelePeriod.
https://tasmota.github.io/docs/Smart-Meter-Interface/#meter-metrics

**Comment by Steve2901** [comment]
Also es wäre nicht schlimm wenn der ESP im Sekundentakt die Daten vom Zähler erfasst, was ja gut ist, denn wenn daten Fließen dann soll der ESP diese "verarbeiten". Es geht mir eigentlich um das Sendeinvervall zum MQTT Server. Da fehlt mir irgendwie eine Option wo man einstellen kann, dass erst nach einer gewissen Zeit die Daten gesendet werden. Also wenn ich 3 Sekunden einstelle, dann soll alles was vor den 3 Sekunden empfangen wurde verworfen werden und nur der Wert, der bei der 3 Sekunde im Speicher des ESP liegt, an den MQTT Server gesendet werden. Ich denke das wäre kein riesiger Aufwand sowas zu implementieren. Ich muss meinen MQTT Server ja nicht ständig zubomben bzw ich brauch nicht im ioBroker im Sekunden-Takt die Daten vom MQTT server. Wäre echt super wenn sowas zum einstellen gäbe.... Ist sowas machbar?

So it wouldn't be a bad thing if the ESP records the data from the counter every second, which is a good thing, because if data is flowing then the ESP should "process" it. I am actually concerned with the transmission interval to the MQTT server. I'm missing an option where you can set that the data is only sent after a certain time. So if I set 3 seconds, then everything that was received before the 3 seconds should be discarded and only the value that is in the ESP's memory at 3 seconds should be sent to the MQTT server. I think this would not be a huge effort to implement. I don't have to constantly bombard my MQTT server or I don't need the data from the MQTT server in the ioBroker every second. Would be really great if there was something like this to set.... Is this feasible?

**Comment by Steve2901** [comment]
Soweit ich weiß kann man das bei der Software
"AI-on-the-edge-device" auch machen (Software für den ESP um Analoge Zähler zu digitalisieren). Da kann man einstellen in welchem Intervall er den Zähler auslesen soll, die Daten verarbeiten soll und dann senden soll. Sowas wäre in Tasmota bei MQTT super :)

As far as I know, you can do this with the software
"AI-on-the-edge-device" software (software for the ESP to digitise analogue meters). You can set the interval at which it should read the counter, process the data and then send it. Something like that would be great in Tasmota with MQTT :)

![grafik](https://github.com/user-attachments/assets/10900703-1b4f-4e5b-823b-93d12eb5718c)

**Comment by sfromis** [comment]
TelePeriod is not relevant to those messages. And Tasmota is not *reading* from the meter, instead it is the meter which is sending when it wants. Did you look into the link I posted about what the Smart Meter Interface publishes?

**Comment by HofeBY** [comment]
probier mal
https://github.com/HofeBY/Tasmota-Firmware
und ließ die Liesmich-zuerst.txt
Telemetrieperiode habe ich bei mir auf 10(sec) gestellt und das funktioniert bei meiner FW compilation auch so.
Wie sieht die script.txt aus ?

**Reply by HofeBY** [reply]
Als script.txt würde ich folgendes für GPIO3 probieren:

>D
>B
->sensor53 r
>M 1
+1,3,s,1,9600,MT175
1,77070100010800ff@1000,1.8.0 Bezug,kWh,Total_in,1
1,77070100020800ff@1000,2.8.0 Lieferung,kWh,Total_out,1
1,77070100100700ff@1,16.7.0 akt.Leistung,W,Power_cur,1
1,77070100240700ff@1,L1 akt.Leistung,W,Power_p1,1
1,77070100380700ff@1,L2 akt.Leistung,W,Power_p2,1
1,770701004C0700ff@1,L3 akt.Leistung,W,Power_p3,1
1,77070100000009ff@#,Server_ID,,Server_ID,0
#

**Reply by Steve2901** [reply]
Hi HofeBy, Danke für dein Feedback. 10 Sekunden ist eigentlich weng zu lang, ich dachte da eher an 3 Sekunden so wie ich es auch im ioBroker einstellen kann.....

Aer ich probiere das mal am WE aus, bin unter der Woche recht voll.

Aber ich versteht echt nicht (soll kein Vorwurf sein) wieso man nicht einfach sagen kann, ich sende nach Zeitspanne x die Daten an den MQTT Server. Das wäre wirllich das einfachste und würde auch alle Probleme lösen.....

**Reply by HofeBY** [reply]
@Steve2901
Ich bin auch nur Anwender. Frage geht also an Entwickler.
Die minimale Transferfrequenz via mqtty finde ich eigentlich zwecks Updatefrequenz & geringere Netzbelastung OK.
Die Zähler- und Sensor-werte braucht´s wirklich nicht öfter als alle 10sec.
Da kannste 10 Leute fragen und bekommst 15 Wünsche.
 
Man kann ja auch nur einen bestimmten Wert öfters übermitteln.
z.B. Zeile für 16.7.0 akt.Leistung
1,77070100100700ff@1,16.7.0 akt.Leistung,W,Power_cur,17

Zum Beobachten der Updatefrequenz und welche Daten übertragen werden LOG einsehen.
